### PR TITLE
Reprogram the Teensy from the WALT app

### PR DESCRIPTION
**Pull request by BenjaminFair**
This makes getting started and dealing with changes to the firmware much easier. I think it's very important for general usability.

### CODE REVIEW

**Comment by kamrik** [comment]
Looks good. Thanks!
Merging.

File: android/WALT/app/src/main/java/org/chromium/latency/walt/ClockManager.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
 public class ClockManager {
 
     static final int TEENSY_VID = 0x16c0;
+    static final int TEENSY_PID = 0x0485; // TeensyLC only
```
Comment (gkasten) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/ClockManager.java]: add reference

File: android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/Connection.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    public void write(byte[] buf, int timeout) {
+        write(buf, 0, buf.length, timeout);
+    }
+
+    public void write(byte[] buf, int index, int len, int timeout) {
+        if (!isConnected()) return;
+
+        while (timeout > 0) {
+            // USB HID Set_Report message
+            int result = mUsbConnection.controlTransfer(0x21, 9, 0x0200, index, buf, len, timeout);
```
Comment (gkasten) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/Connection.java]: comment the magic numbers

File: android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/Programmer.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.chromium.latency.walt.programmer;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.chromium.latency.walt.R;
```
Comment (gkasten) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/Programmer.java]: arggh I thought these were not necessary

File: android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/BootloaderConnection.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package org.chromium.latency.walt.programmer;
```
Comment (gkasten) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/programmer/BootloaderConnection.java]: copyright header

**Comment by gkasten** [comment]
BootloaderConnection.java - copyright
comment the magic numbers in  mUsbConnection.controlTransfer
I thought import org.chromium.latency.walt.R; was not necessary
add reference for TEENSY_PID and _VID

**Pull request merged**